jenkins/scripts/select-compiler.sh

```diff
@@ -4,14 +4,14 @@ if [ "$DONTSELECT_COMPILER" != "DONT" ]; then
   NODE_NAME=${NODE_NAME:-$HOSTNAME}
   echo "Selecting compiler based on $NODE_NAME"
   case $NODE_NAME in
-    *ppc64_le* ) SELECT_ARCH=PPC64LE ;;
+    *ppc64*le* ) SELECT_ARCH=PPC64LE ;;
     *s390x* ) SELECT_ARCH=S390X ;;
     *aix* ) SELECT_ARCH=AIXPPC ;;
   esac
 fi
 
 # get node version
-if [ -z ${NODEJS_MAJOR_VERSION+x} ]; then
+if [ -z "$NODEJS_MAJOR_VERSION" ]; then
```

Comment:
Wondering about this change.

Reply:
The previous code wasn't working for me, this does.  Possibly a shell difference? I am not at all sure why there was such a convoluted way of checking of the env var is set, this seems to work better:
```
core/build (centos7-ppc-support % u=) % if [ -z ${ZZZ+x} ]; then echo T; fi
T
core/build (centos7-ppc-support % u=) % if [ -z "$ZZZ" ]; then echo T; fi
T
core/build (centos7-ppc-support % u=) % env | grep ZZZ
```

Tomorrow I can revert it and post the error caused by the original line.

Reply:
This change will match both an empty string and an unset environment variable. I think this change is OK because if it's empty then it's still useless and should be re-generated, but it's a bit weird that it's necessary. In what context is `NODEJS_MAJOR_VERSION` not set? Are there Jenkins jobs that aren't using the versioning plugin?

Reply:
> 
> 
> This change will match both an empty string and an unset environment variable. I think this change is OK because if it's empty then it's still useless and should be re-generated, but it's a bit weird that it's necessary. In what context is `NODEJS_MAJOR_VERSION` not set? Are there Jenkins jobs that aren't using the versioning plugin?

Yes. Anything that isn't building Node.js from source but still needs to do compilation (i.e. natve addons), e.g. [citgm-smoker-nobuild](https://ci.nodejs.org/view/Node.js-citgm/job/citgm-smoker-nobuild/configure). [node-test-node-addon-api](https://ci.nodejs.org/view/x%20-%20Abi%20stable%20module%20API/job/node-test-node-addon-api/configure).